### ISSUE TITLE
Improving general performance

### ISSUE THREAD

**Issue by ErikBjare** [ActivityWatch/aw-webui#124]
This was enlightening: https://checklyhq.com/blog/2018/08/how-we-got-a-100-lighthouse-performance-score-for-our-vue.js-app/

**Comment by ErikBjare** [comment]
Might have been fixed to an extent by #123

**Comment by xylix** [comment]
I did some research since I've found the 1-3second load times on localhost annoying, especially for large datasets. Do you think the problem is server- not frontend side for large (1+ months) buckets?

Anyway, the case is that having 1 a month bucket loaded does cause performance issues and CPU hogging. There's a "WARNING: infinite loop in redraw?" on the web console when viewing timeline which could be related.

I started following the instructions anyway for the heck of it and webpack-bundle-analyzer revealed aw-webui does load quite a bit of JS:

<img width="228" alt="Screenshot 2020-02-01 at 12 10 23" src="https://user-images.githubusercontent.com/13387304/73590501-e187d400-44eb-11ea-880f-82fb9331f195.png">
<img width="1275" alt="Screenshot 2020-02-01 at 12 09 44" src="https://user-images.githubusercontent.com/13387304/73590502-e3ea2e00-44eb-11ea-8fca-6a1dc2c687b0.png">

Lighthouse revealed some accessibility and loadtime issues on slow networks but slow network conditions is a mostly irrelevant problem when serving locally.

[activitywatch-lighthouse.zip](https://github.com/ActivityWatch/aw-webui/files/4142422/activitywatch-lighthouse.zip)

@ErikBjare By the way, is the mobile version available somewhere?

**Comment by ErikBjare** [comment]
@xylix The Android app is developed in https://github.com/ActivityWatch/aw-android

Having a large JS bundle shouldn't be that much of an issue since it's served locally, but parsing JS can still be quite slow so might introduce some slowness in initially loading the web UI, but not when it comes to waiting for queries to complete.

Almost all performance issues relating to large datasets are due to the server being slow to process the queries. aw-server-rust is much much faster in this regard, and since we're planning on migrating to it in the near/mid-term it'll solve itself. 

The Android app already uses aw-server-rust so there things aren't as bad on the server-side performance-wise, but since it also generates far more events (since every activity change generates an event) and runs on slower hardware it can sometimes feel slow as well.

**Comment by xylix** [comment]
Okay cool. I'll try looking into updating to vis-timeline.

Thanks for the link to the rust-server progress. Seems like it's going to be a great upgrade.

**Comment by ErikBjare** [comment]
The reason for there not being any UI tests is that we don't have many UI related bugs and maintaining UI tests is a lot of work. I'd like to keep it that way for as long as possible. 

It's pretty easy to manually check that everything works as intended, so automating it doesn't have any significant benefit.

**Comment by johan-bjareholt** [comment]
This is an never ending issue, could we please split it into new separate issues such as "lowering bundle size" or "audit web-ui performance"?

**Comment by xylix** [comment]
Good idea @johan-bjareholt , I agree we should split this into better actionable chunks
